### PR TITLE
Adding a test to people nearby search method and sort people by recent signups  [WIP]

### LINKED ISSUE #3729: Search API: endpoint /srch/nearbyPeople needs to sort by most recent users
As discussed here ([Question: What more data-layers can we show on map?](https://publiclab.org/questions/sagarpreet/07-01-2018/what-more-data-layers-can-we-show-on-map)) I implemented (#3719) a new method to search nearby people with respect to given latitude, longitude and tag (optional). 

Code is in here: https://github.com/publiclab/plots2/blob/master/app/services/search_service.rb#L118-L146

```
  def tagNearbyPeople(query, tag, limit = 10)
    raise("Must separate coordinates with ,") unless query.include? ","

    lat, lon =  query.split(',')

    user_locations = User.where('rusers.status <> 0')\
                         .joins(:user_tags)\
                         .where('value LIKE ?', 'lat:' + lat[0..lat.length - 2] + '%')\
                         .distinct
    if tag.present?
      user_locations = User.joins(:user_tags)\
                       .where('user_tags.value LIKE ?', tag)\
                       .where(id: user_locations.select("rusers.id"))
    end

    ids = user_locations.collect(&:id).uniq || []

    items = User.where('rusers.status <> 0').joins(:user_tags)
                .where('rusers.id IN (?) AND value LIKE ?', ids, 'lon:' + lon[0..lon.length - 2] + '%')
    
    # selects the items whose node_tags don't have the location:blurred tag
    items.select do |item|
      item.user_tags.none? do |user_tag|
        user_tag.name == "location:blurred"
      end
    end
      
    items = items.limit(limit)
  end
```
Now, as a next step we need to modify the method to sort the results by most recent sign up at least.

The [`search_profiles()`](https://github.com/publiclab/plots2/blob/master/app/services/search_service.rb#L32
) method sorts the results by most recent users, I think the same strategy can be used here.

### PR DESCRIPTION
**Pull request by Nada1996**
Fixes #3729  #3766  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Nada1996** [comment]
@jywarren @milaaraujo I added a test to the tagNearbyPeople function, also it now sorts the results by the most recent signups by default.

Now, I want to pass the "order_by" as a param to sort it by followers or recent logins for example, but  I want some feedback. 
Thanks in advance

**Comment by Nada1996** [comment]
@milaaraujo Thanks for your feedback, I working on it now adjusting the code then I'll push the code to the PR right away 👍

**Comment by milaaraujo** [comment]
Great, `tagNearbyPeople()` method looks good!

Now, about the tests, you still need to do the assertions to compare what you were expecting and what the endpoint really returns. Take  look at [this test for example](https://github.com/publiclab/plots2/blob/master/test/functional/api/search_api_test.rb#L11-L33):

>assert_equal "/profile/steff3",   json['items'][0]['doc_url']
assert_equal "/profile/steff2",   json['items'][1]['doc_url']
assert_equal "/profile/steff1", json['items'][2]['doc_url']

You will need to use the data in https://github.com/publiclab/plots2/tree/master/test/fixtures. Let me know if you need any help with this.

**Comment by Nada1996** [comment]
@milaaraujo I added locations tags to existing users and adjusted the code accordingly,
I also added some asserts to check on the sorting functionality and the search by tag.
Please let me know if there's anything else that needs to be done.
Thanks in advance

**Comment by milaaraujo** [comment]
This is looking great! I just added a small review. Thank you for your work here. :smile:

What do you think about it, @jywarren?

File: app/services/search_service.rb [hunk 1]
```diff
@@ -135,14 +135,19 @@ def tagNearbyPeople(query, tag, limit = 10)
...
+    end
     # selects the items whose node_tags don't have the location:blurred tag
     items.select do |item|
       item.user_tags.none? do |user_tag|
         user_tag.name == "location:blurred"
       end
     end
 
-    items = items.limit(limit)
+    items = items.order(id: :desc).limit(limit)
```
[the remark this answers is not in the document]
Comment (jywarren) [reply app/services/search_service.rb]: This is fantastic. It's really readable and well implemented, and with a great test. Just this one issue to resolve and it's ready to merge!

**Comment by milaaraujo** [comment]
Thank you, @Nada1996!

**Comment by jywarren** [comment]
This looks super. Fantastic work, and great teamwork from @milaaraujo as well!

**Pull request merged**